### ISSUE TITLE
prefer-promise-reject-errors: false negative for spread-argument reject calls `reject(...args)`

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25620]
## Summary

oxlint's native `prefer-promise-reject-errors` rule fails to flag a Promise rejection whose argument is a spread element, e.g. `reject(...args)`. ESLint's core rule (both v7 and v9) reports this construct; oxlint reports nothing. The rule does fire on literal non-Error rejections like `reject('a string error')`, so the gap is specific to the spread-call form. The recent changes to this rule in 1.77 (#25238, ambient error variables) and 1.78 (#25378, parenthesized calls) do not affect this case.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.js`:

```js
new Promise((resolve, reject) => {
	somePromise.catch((...args) => {
		reject(...args) // should be flagged: rejection reason is not provably an Error
	})
})
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {
		"prefer-promise-reject-errors": ["error", {"allowEmptyReject": false}]
	}
}
```

Run:

```
oxlint -c .oxlintrc.json repro.js
```

## Expected

A `prefer-promise-reject-errors` diagnostic on the `reject(...args)` call. ESLint 7.23.0 and ESLint 9.39.4 (core rule, same options) both report:

```
repro.js
  3:3  error  Expected the Promise rejection reason to be an Error  prefer-promise-reject-errors
```

The spread argument means the rejection reason is not a known Error, which is exactly what this rule is meant to catch (it already fires on literal non-Error rejections like `reject('a string error')`).

## Actual

oxlint 1.78.0 emits no diagnostic and exits 0:

```
$ npx oxlint -c .oxlintrc.json repro.js
$ echo $?
0
```

A control file with a literal non-Error rejection on line 2 is flagged, confirming the rule is active with this config:

```
control.js:2:2: error eslint(prefer-promise-reject-errors): Expected the Promise rejection reason to be an Error
```

## Cross-checked

| Engine                          | Version | `reject(...args)`  |
| ------------------------------- | ------- | ------------------ |
| ESLint (core rule)              | 7.23.0  | flagged (3:3)      |
| ESLint (core rule, flat config) | 9.39.4  | flagged (3:3)      |
| oxlint (native rule)            | 1.78.0  | not flagged        |
